authentication/views.py

```diff
@@ -275,3 +281,63 @@ def checkConfiguration(request):
     return Response(
         status=status.HTTP_200_OK, data={"errors": errors} if errors else {}
     )
+
+
+def github_login(request):
+    client_id = secrets.get_secret("GITHUB_CLIENT_ID")
+    return redirect(f"https://github.com/login/oauth/authorize?client_id={client_id}")
+
+
+class GitHubLoginCallbackView(LoginView):
+    @staticmethod
+    def validate_and_return_user(request):
+        code = request.GET.get("code")
+        if code:
+            url = "https://github.com/login/oauth/access_token"
+            headers = {"Accept": "application/json"}
+            params = {
+                "client_id": secrets.get_secret("GITHUB_CLIENT_ID"),
+                "client_secret": secrets.get_secret("GITHUB_CLIENT_SECRET"),
+                "code": code,
+                "redirect_uri": "http://localhost/api/auth/github-callback",
+            }
+            response = requests.post(url, params=params, headers=headers)
+            response_data = response.json()
+            access_token = response_data.get("access_token")
+            if access_token:
+                headers = {
+                    "Authorization": f"Bearer {access_token}",
+                    "Accept": "application/json",
+                }
+                response = requests.get("https://api.github.com/user", headers=headers)
+                user_data = response.json()
+                user_name = user_data.get("login")
+                user_email = user_data.get("email")
+                logging.info(f"received userinfo from github: {response}")
```

Comment:
have you at least read which is the output of this log? cause in the case you really had, you would have understood that this cannot be the proper variable to use.

Plus, are we sure that we can log this user info? cause there is sensitive information. I would just use a `debug` log for debugging purposes because that is the reason why this log can become useful to us. To understand what is happening while we are developing